### PR TITLE
Fix gr.Warning/gr.Info when using generators + interface

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

To test, run this demo

```python
import gradio as gr
import time

def greet(name):
    gr.Info("Starting...")
    time.sleep(2)
    for i, letter in enumerate(name):
        if letter == " ":
            gr.Warning(f"Space detected {i}!")
        yield "Hello " + name[:i+1] + "!"
        time.sleep(0.5)
    time.sleep(2)

demo = gr.Interface(fn=greet, inputs="text", outputs="text")
demo.queue(concurrency_count=10).launch()
```

Also added some e2e tests for info and warning as well.

This is fixing a bug for an unreleased feature, so I don't think we need a changelog entry here.

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests & Changelog

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

1. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`
  
1. Unless the pull request is labeled with the "no-changelog-update" label by a maintainer of the repo, all pull requests must update the changelog located in `CHANGELOG.md`:

Please add a brief summary of the change to the Upcoming Release section of the `CHANGELOG.md` file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

### CODE REVIEW

File: test/test_blocks.py [hunk 1]
```diff
@@ -347,43 +347,33 @@ async def wrapped():
...
+                await asyncio.sleep(0.2)
+
+        demo = gr.Interface(async_iteration, gr.Number(precision=0), gr.Number()).queue()
+        outputs = []
+        with connect(demo) as client:
+            for output in client.submit(3, api_name="/predict"):
+                outputs.append(output)
+        assert outputs == [0, 1, 2]
+
+    def test_sync_generators(self, connect):
```
Comment (freddyaboulton) [inline test/test_blocks.py:365]: Refactored this test to use `connect`

File: test/test_blocks.py [hunk 2]
```diff
@@ -347,43 +347,33 @@ async def wrapped():
             await asyncio.gather(_get_ws_pred(3, 0)(), _get_ws_pred(4, 1)())
         finally:
             demo.close()
+    
+    def test_async_generators_interface(self, connect):
```
Comment (freddyaboulton) [inline test/test_blocks.py:351]: Added a test for async generator + interface case here

File: js/app/test/blocks_chained_events.spec.ts [hunk 3]
```diff
@@ -4,20 +4,24 @@ import type { Response } from "@playwright/test";
 test(".success event runs after function successfully completes. .success should not run if function fails", async ({
 	page
 }) => {
+	let last_iteration;
```
Comment (freddyaboulton) [inline js/app/test/blocks_chained_events.spec.ts:7]: Just rewrote these tests to use websockets since queue is enabled

**Comment by aliabid94** [comment]
Looks and works great, thanks @freddyaboulton for getting this in!

**Comment by abidlabs** [comment]
Let's merge this in so that we can fully test #4742

**Pull request merged**